### PR TITLE
highlight now shows highlight description instead of project descript…

### PR DESCRIPTION
**Pull request by niraymak**
…ion.

## Description

A few sentences describing the overall goals of the pull request's commits.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

create highlight
update highlight

## Link to issue

Closes: #245

### CODE REVIEW

**Comment by Brend-Smits** [comment]
Did you test this @niraymak?

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
It looks good, although I think you missed a few important pieces. 
1. Adding a highlight is possible for certain user groups (like PR/Admin roles). See the project/modal-highlight component. We will need to add a text field in this modal as well so that new highlights are added with a description. 
2. The models that we use in the frontend (models/domain/highlight.ts but also under models/resources/highlight-add&update) will need to be updated to represent this change. 

Let me know how it goes!

**Comment by niraymak** [comment]
Functionality mentioned by Brend works now. Styling still has to be fixed. Also I want to look into a potential bug. I will do that later.

**Comment by niraymak** [comment]
It is also possible to add a description now. 
When deleting a highlight the description shows up aswell.

Please review code.

File: src/app/models/domain/hightlight.ts [hunk 1]
```diff
@@ -23,5 +23,6 @@ export interface Highlight {
   project: Project;
   startDate?: string;
   endDate?: string;
+  description?: string;
```
Comment (Brend-Smits) [inline src/app/models/domain/hightlight.ts:26]:
```suggestion
  description: string;
```
The description should not be nullable

File: src/app/models/resources/highlight-add.ts [hunk 2]
```diff
@@ -19,4 +19,5 @@ export interface HighlightAdd {
   projectId: number;
   startDate?: Date;
   endDate?: Date;
+  description?: string;
```
Comment (Brend-Smits) [inline src/app/models/resources/highlight-add.ts:22]:
```suggestion
  description: string;
```
The description should not be nullable

File: src/app/modules/project/modal-highlight-delete/modal-highlight-delete.component.html [hunk 3]
```diff
@@ -39,9 +39,10 @@ <h2>Delete highlight</h2>
                 <div class="highlight-wrapper">
                   <input type="checkbox" id="checkbox" [formControlName]="highlight.id" class="" />
                   <label for="checkbox">
-                    <div class="row no-gutters">
-                      <div class="col-6"><strong>Start date:</strong>{{highlight.startDate}}</div>
-                      <div class="col-6"><strong>End date:</strong>{{highlight.endDate}}</div>
+                    <div class="row">
+                      <div class="col-4"><strong>Start date:</strong>{{highlight.startDate}}</div>
+                      <div class="col-4"><strong>End date:</strong>{{highlight.endDate}}</div>
+                      <div class="col-4"><strong>Description:</strong>{{highlight.description}}</div>
```
Comment (Brend-Smits) [inline src/app/modules/project/modal-highlight-delete/modal-highlight-delete.component.html:45]: Let's revert this change, after looking at the modal I don't think this will really give the user more information. It will make the modal look messy (especially with longer descriptions) and will not serve much of a purpose at this moment.

File: src/app/modules/project/modal-highlight/modal-highlight.component.ts [hunk 4]
```diff
@@ -87,6 +90,10 @@ export class ModalHighlightComponent {
       this.dateErrorMessage = 'Error: Start date can\'t be later than end date';
       return;
     }
+    if (highlightFormResult.description === null) {
+      this.dateErrorMessage = 'Error: Please enter a description';
+      return;
+    }
```
Comment (Brend-Smits) [inline src/app/modules/project/modal-highlight/modal-highlight.component.ts:96]:
```suggestion
    } else {
      if (highlightFormResult.description.length < 75 || highlightFormResult.description.length > 155)
      {
        this.dateErrorMessage = 'Error: Description must be between 75 and 155 characters long';
        return;
      }
    }
```

Add additional check and make sure that we don't receive too short/long descriptions.

File: src/app/modules/project/modal-highlight/modal-highlight.component.scss [hunk 5]
```diff
@@ -87,3 +89,17 @@
...
+//   box-shadow: 0px 3px 6px #0000001a;
+//   height: 45px;
+//   width: 100%;
+//   margin-bottom: 25px;
+//   padding: 0px 15px;
+// }
+
+// input:focus {
+//   outline: auto #e4e4e4;
+// }
```
Comment (Brend-Smits) [inline src/app/modules/project/modal-highlight/modal-highlight.component.scss:105]: Remove commented out code

File: src/app/modules/project/modal-highlight/modal-highlight.component.html [hunk 6]
```diff
@@ -37,6 +37,12 @@ <h3>End Date</h3>
...
           </div>
         </div>
+        <div class="col-12">
+          <div>
+            <h3>Project description</h3>
+            <input type="text" placeholder="Project description" formControlName="description">
+          </div>
+        </div>
         <div class="col-12" *ngIf="dateErrorMessage != null">
           <p class="date-error-message">{{dateErrorMessage}}</p>
```
Comment (Brend-Smits) [inline src/app/modules/project/modal-highlight/modal-highlight.component.html:47]: Since we're now using this "date error message" for more than just the date, it's best to rename it to something like "validation error message". 
When doing this, make sure to also rename the .scss class.

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
Good work, we're nearly there. A few more touch-ups here and there and then it's ready to be merged. 
Keep it up 👍🏼

File: src/app/modules/project/modal-highlight-delete/modal-highlight-delete.component.html (hunk 3 above)
Comment (Brend-Smits) [reply src/app/modules/project/modal-highlight-delete/modal-highlight-delete.component.html]: This file was not reverted properly.  Please check the file history and revert back to the correct version.

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
I merged Develop into this PR to fix conflicts, as well as to fix any possible issues with line endings. 
Before I approve this PR and we can merge it, fix the last open comment.

**Review by Brend-Smits — DISMISSED** [review]

**Review by devoncavis — APPROVED** [review]
Everything appears to be working for me and the code looks well done according to the guidelines.

**Review by Brend-Smits — APPROVED** [review]
